heron/common/src/cpp/network/network_error.h

```diff
@@ -30,4 +33,12 @@ enum NetworkErrorCode {
   TIMEOUT
 };
 
+static inline std::ostream& operator<<(std::ostream& os, const NetworkErrorCode code) {
+  std::string strs[] = { "OK", "Connection error", "Read error", "Write error",
```

Comment:
Can these instead be bound directly to the enum values themselves, instead of managed separately?
http://en.cppreference.com/w/cpp/language/enum